File: POO/Constructores_Clases/Class/GO/Estudiante.go
package main

import "fmt"

type Estudiante struct {
    nombre string
    edad   int
    grado  string
    promedio float64
}

// Constructor 1: Solo nombre y edad
func NewEstudiante(nombre string, edad int) *Estudiante {
    return &Estudiante{
        nombre: nombre,
        edad:   edad,
        grado:  "No asignado",
        promedio: 0.0,
    }
}

// Constructor 2: Todos los parámetros
func NewEstudianteCompleto(nombre string, edad int, grado string, promedio float64) *Estudiante {
    return &Estudiante{
        nombre:   nombre,
        edad:     edad,
        grado:    grado,
        promedio: promedio,
    }
}

func (e *Estudiante) Mostrar() {
    fmt.Printf("Estudiante: %s, Edad: %d, Grado: %s, Promedio: %.2f\n",
        e.nombre, e.edad, e.grado, e.promedio)
}

func main() {
    est1 := NewEstudiante("Ana", 20)
    est2 := NewEstudianteCompleto("Carlos", 22, "Ingeniería", 8.5)
    
    est1.Mostrar()
    est2.Mostrar()
}

File: POO/Constructores_Clases/Class/JAVA/Estudiante.java
public class Estudiante {
    private String nombre;
    private int edad;
    private String grado;
    private double promedio;

    // Constructor 1: Solo nombre y edad
    public Estudiante(String nombre, int edad) {
        this.nombre = nombre;
        this.edad = edad;
        this.grado = "No asignado";
        this.promedio = 0.0;
    }

    // Constructor 2: Todos los parámetros
    public Estudiante(String nombre, int edad, String grado, double promedio) {
        this.nombre = nombre;
        this.edad = edad;
        this.grado = grado;
        this.promedio = promedio;
    }

    public void mostrar() {
        System.out.printf("Estudiante: %s, Edad: %d, Grado: %s, Promedio: %.2f%n",
            nombre, edad, grado, promedio);
    }

    public static void main(String[] args) {
        Estudiante est1 = new Estudiante("Ana", 20);
        Estudiante est2 = new Estudiante("Carlos", 22, "Ingeniería", 8.5);

        est1.mostrar();
        est2.mostrar();
    }
}